src/Text.Analyzers/Core/CodeAnalysisDictionary.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Text.Analyzers
+{
+    /// <summary>
+    /// Source for "recognized" misspellings and "unrecognized" spellings obtained by parsing either
+    /// XML or DIC code analysis dictionaries.
+    /// </summary>
+    /// <Remarks>
+    /// <seealso href="https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-customize-the-code-analysis-dictionary?view=vs-2019"/>
+    /// </Remarks>
+    internal class CodeAnalysisDictionary
+    {
+        protected CodeAnalysisDictionary(XDocument document)
```

Comment:
This constructor feels weird. I think it would be better if all the processing/loading happened in the static factory methods on `CodeAnalysisDictionary` and the constructors only accept the final immutable data. This is the pattern used everywhere across this repo.

Reply:
Also why are these constructors all protected?

Reply:
Done and removed protected